corehq/apps/locations/bulk_management.py

```diff
@@ -421,7 +431,11 @@ def bulk_commit(self, type_stubs, location_stubs):
             loc.lookup_old_collection_data(self.old_collection)
 
         type_objects = save_types(type_stubs, self.excel_importer)
-        save_locations(location_stubs, type_objects, self.domain, self.excel_importer)
+        types_changed = any(loc_type.needs_save for loc_type in type_stubs)
+        moved_to_root = any(loc.moved_to_root for loc in location_stubs)
+        delay_updates = not (types_changed or moved_to_root)
```

Comment:
Does this PR make any improvements for situations where this expression evaluates to false? If not, how often does that happen? If often, maybe we should look harder at moving to adjacency list + recursive queries? In case you haven't read the articles linked by @snopoke, we do not need to change the data model to do that; we could skip all the MPTT updates and just put bogus values in `lft` and `rght`.